src/worker.cpp

```diff
@@ -651,12 +652,65 @@ bool Worker::isCPUIOGoodOnly(const std::string& pgKeyword)
     return false;
 }
 
-void Worker::primeInventory(const types::IPZVpdMap& ipzVpdMap,
-                            types::ObjectMap primeObjects)
+bool Worker::primeInventory(const std::string& i_vpdFilePath)
 {
-    // TODO implementation
-    (void)ipzVpdMap;
-    (void)primeObjects;
+    logging::logMessage("primeInventory called for FRU " + i_vpdFilePath);
+
+    if (i_vpdFilePath.empty())
+    {
+        logging::logMessage("Empty VPD file path given");
+        return false;
+    }
+
+    if (m_parsedJson.empty())
+    {
+        logging::logMessage(
+            "A config JSON is required to prime an inventory for a FRU " +
+            i_vpdFilePath);
+        return false;
+    }
+
+    types::ObjectMap l_objectInterfaceMap;
+    for (const auto& l_aFru : m_parsedJson["frus"][i_vpdFilePath])
+    {
+        types::InterfaceMap l_interfaces;
+        sdbusplus::message::object_path l_fruObjectPath(
+            l_aFru["inventoryPath"]);
+
+        // Add extra interfaces mentioned in the Json config file
+        if (l_aFru.contains("extraInterfaces"))
+        {
+            populateInterfaces(l_aFru["extraInterfaces"], l_interfaces,
+                               types::IPZVpdMap{});
+        }
+
+        // Process FRUS which are embedded in the parent FRU and whose VPD
+        // will be synthesized.
+        if ((l_aFru.value("embedded", true)) &&
+            (!l_aFru.value("synthesized", false)))
+        {
+            processEmbeddedAndSynthesizedFrus(l_aFru, l_interfaces);
```

Comment:
Going by the flow here, you are marking present as true. Where are you marking present as false?
Ideally, priming is done when FRU is not present.